packages/cloud_firestore_odm_generator/lib/src/collection_data.dart

```diff
@@ -425,30 +425,25 @@ extension on ClassElement {
   Iterable<VariableElement> allFields({
     required bool hasFreezed,
     required List<ConstructorElement> freezedConstructors,
-  }) sync* {
+  }) {
     if (hasFreezed) {
-      yield* freezedConstructors.single.parameters;
+      return freezedConstructors.single.parameters;
     } else {
       final uniqueFields = <String, FieldElement>{};
 
-      for (final field in fields) {
-        if (field.getter != null && !field.getter!.isSynthetic) {
-          continue;
-        }
-        uniqueFields[field.name] ??= field;
-      }
+      final allFields = [
+        ...fields,
+        ...allSupertypes
+            .where((e) => !e.isDartCoreObject)
+            .expand((e) => e.element.fields),
+      ];
```

Comment:
Could you use `Iterable.followedBy`? To avoid allocating an array just for looping over fields

Reply:
I tried but I get type errors such as

`type 'ExpandIterable<InterfaceType, FieldElement>' is not a subtype of type 'Iterable<FieldElementImpl>' of 'other'`

`type 'CastIterable<FieldElement, FieldElement>' is not a subtype of type 'Iterable<FieldElementImpl>' of 'other'`

Reply:
You can do `const <FieldElement>[].followedBy(fields).followedBy(...)`

Reply:
Isn't that instantiating an array which you wanted to avoid?

Reply:
Not really. The only "array" here is a constant empty one.
That's different from an array created at runtime with a carrying length